test/test_datapipe.py

```diff
@@ -1149,20 +1149,30 @@ def _helper(ref_fn, fn, input_col=None, output_col=None):
             for constr in (list, tuple):
                 datapipe = dp.iter.IterableWrapper([constr((0, 1, 2)), constr((3, 4, 5)), constr((6, 7, 8))])
                 res_dp = datapipe.map(fn, input_col, output_col)
-                ref_dp = datapipe.map(ref_fn)
+                ref_dp = datapipe.map(ref_fn) if ref_fn is not None else datapipe
                 self.assertEqual(list(res_dp), list(ref_dp))
                 # Reset
                 self.assertEqual(list(res_dp), list(ref_dp))
 
+        p_fn_n1 = partial(fn_n1, d1=1)
+
+
         # Replacing with one input column and default output column
         _helper(lambda data: (data[0], -data[1], data[2]), fn_11, 1)
         _helper(lambda data: (data[0], (-data[1], data[1]), data[2]), fn_1n, 1)
+
+        _helper(lambda data: (data[0], data[1], data[0] + data[1]),
+                lambda d0, d1: d0 + d1, (0, 1), 2)
+        _helper(lambda data: (data[0], data[1], 1 + data[1]), p_fn_n1, 1, 2)
         # The index of input column is out of range
         with self.assertRaises(IndexError):
             _helper(None, fn_1n, 3)
+
         # Unmatched input columns with fn arguments
         with self.assertRaises(TypeError):
             _helper(None, fn_n1, 1)
+            _helper(None, lambda d0, d1: d0 + d1, 0)
+            _helper(None, p_fn_n1, (0, 1))
```

Comment:
You might need to change the expected Error after it's changed to `ValueError`.

Reply:
Also resolved in https://github.com/pytorch/pytorch/pull/79344/commits/ef6385f2f5de044013136f865d066eebe1bbc2b1